partiql-spi/src/main/kotlin/org/partiql/spi/fn/builtins/internal/AccumulatorDistinct.kt

```diff
@@ -1,24 +1,17 @@
 package org.partiql.spi.fn.builtins.internal
 
 import org.partiql.eval.value.Datum
-import org.partiql.value.PartiQLValue
-import org.partiql.value.PartiQLValueExperimental
 import java.util.TreeSet
 
-@OptIn(PartiQLValueExperimental::class)
 internal class AccumulatorDistinct(
     private val _delegate: Accumulator,
 ) : Accumulator() {
 
-    // TODO: Add support for a datum comparator once the accumulator passes datums instead of PartiQL values.
-    @OptIn(PartiQLValueExperimental::class)
-    private val seen = TreeSet(PartiQLValue.comparator())
+    private val seen = TreeSet(Datum.comparator())
```

Comment:
Yeah, I'll add a comment. But, it comes down to the lack of support for hashCode and equals. The reason we can't support them outright for aggregations/grouping is because of how the PartiQL Specification dictates that the grouping should act similarly to the `<` operator. This operator compares the arithmetic values of INT, BIGINT, etc -- so a simple hashCode/equals is non-trivial. It *may* be fast to coerce values into some larger type specifically for aggregations -- but this requires further research.

So, a TreeSet is necessary since we can pass in the comparator.